src/xz/writer.rs

```diff
@@ -196,23 +196,12 @@ pub struct XzOptions {
     pub filters: Vec<FilterConfig>,
 }
 
-impl Default for XzOptions {
-    fn default() -> Self {
-        Self {
-            lzma_options: LzmaOptions::default(),
-            check_type: CheckType::Crc32,
-            block_size: None,
-            filters: Vec::new(),
-        }
-    }
-}
-
 impl XzOptions {
     /// Create options with specific preset and checksum type.
     pub fn with_preset(preset: u32) -> Self {
         Self {
             lzma_options: LzmaOptions::with_preset(preset),
-            check_type: CheckType::Crc64,
+            check_type: CheckType::default(),
```

Comment:
There is no need to make this implicit. You only add a layer of indirection for users, that want to know what this function is returning. Such changes also could result in a cascading effect that could later bite us. So please leave it at `CheckType::Crc64`.

Reply:
Merci!